app/models/assessment_user_datum.rb

```diff
@@ -68,6 +69,13 @@ def update_latest_submission
       save!
     end # release lock on AUD
     # see: http://dev.mysql.com/doc/refman/5.0/en/innodb-locking-reads.html
+
+    CourseUserDatum.transaction do
+      CourseUserDatum.lock(true).find(course_user_datum_id)
```

Comment:
Small nit but for consistency mention where it's locked as well

Reply:
fixed! Added comment!